hack/with-go-mod.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+#
+# This script is used to coerce certain commands which rely on the presence of
+# a go.mod into working with our repository. It works by creating a fake
+# go.mod, running a specified command (passed via arguments), and removing it
+# when the command is finished. This script should be dropped when this
+# repository is a proper Go module with a permanent go.mod.
+
+set -e
+
+SCRIPTDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+ROOTDIR="$(git -C "$SCRIPTDIR" rev-parse --show-toplevel)"
+
+set -x
+
+tee "${ROOTDIR}/go.mod" >&2 <<EOF
```

Comment:
Clobbering an existing go.mod file without warning might not be ideal. For example, it could go really badly if (as I have done in the past) go.mod is a symlink to vendor.mod.
```suggestion
if [[ -e "${ROOTDIR}/go.mod" ]]; then
    >&2 echo "WARNING: go.mod already exists."
    exec "$@"
fi

set -x 

tee "${ROOTDIR}/go.mod" >&2 <<EOF
```